### PR TITLE
Fix font-awesome font not being cached by the service-worker

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

Fix snackbar not updating when multiple updates come in in quick succession. 
Fix mark-read service trying to mark inexistent documents as read. Also removes a redundant reducer. 
Fix `currentTab` being very shortly set to `undefined` because the child route is activated _before_ the parent route and also emits in `ActivationEnd` event. 
Removes obsolete dependencies (old `redux` used in AngularJS, these were not removed during 1st cleanup iteration because I didn't know if they were needed for the implementation or not). 

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by dianabarsan** [comment]
Mind having a look, @mrsarm ? Thank you!

File: webapp/src/ts/app.component.ts [hunk 1]
```diff
@@ -120,9 +120,21 @@ export class AppComponent {
   }
 
   private setupRouter() {
+    const getTab = (snapshot) => {
+      let tab;
+      do {
+        tab = snapshot.data.tab;
+        snapshot = snapshot.parent;
+      } while (!tab && snapshot && snapshot.parent);
```
Comment (mrsarm) [inline webapp/src/ts/app.component.ts:128]:
```suggestion
      } while (!tab && snapshot?.parent);
```

File: webapp/src/ts/app.component.ts [hunk 2]
```diff
@@ -231,11 +243,8 @@ export class AppComponent {
     this.loadTranslations();
     this.setupDb();
 
-    if (
-      (<any>window).medicmobile_android &&
-      typeof (<any>window).medicmobile_android.getAppVersion === 'function'
-    ) {
-      this.globalActions.setAndroidAppVersion((<any>window).medicmobile_android.getAppVersion())
+    if (window.medicmobile_android && typeof window.medicmobile_android.getAppVersion === 'function') {
```
Comment (mrsarm) [inline webapp/src/ts/app.component.ts:246]:
```suggestion
    if (typeof window.medicmobile_android?.getAppVersion === 'function') {
```

**Review by mrsarm — APPROVED** [review]
LGTM

**Pull request merged**